src/System.Net.Http/tests/FunctionalTests/HPackTest.cs

```diff
@@ -0,0 +1,74 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using System.Diagnostics;
+using System.IO;
+using System.Net.Test.Common;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Xunit;
+using Xunit.Abstractions;
+using System.Data;
+using System.Runtime.InteropServices.ComTypes;
+
+namespace System.Net.Http.Functional.Tests
+{
+    public class HPackTest : HttpClientHandlerTestBase
+    {
+        protected override bool UseSocketsHttpHandler => true;
+        protected override bool UseHttp2 => true;
+
+        public HPackTest(ITestOutputHelper output) : base(output)
+        {
+        }
+
+        const string LiteralHeaderName = "x-literal-header";
+        const string LiteralHeaderValue = "testing 456";
+
+        [Theory]
+        [MemberData(nameof(HeaderEncodingTestData))]
+        public Task HPack_HeaderEncoding(string headerName, string expectedValue, byte[] expectedEncoding)
+        {
+            return Http2LoopbackServer.CreateClientAndServerAsync(
```

Comment:
We don't need to hyper-optimize tests.  I'd suggest making this test async and awaiting this, rather than returning it out.  That way, stack traces and other related diagnostics will include the test name.